Rest of the details (#8)
alan-turing-institute/climate-informatics-2024-ae: process.md

# Overview & Rationale

> Climate Informatics, like many other communities and fields, has software at its heart. Underlying most publications is a novel piece of software playing some critical role, e.g., embodying a model, processing or analysing data, or producing a visualisation. In order for such software artefacts to have the most impact, they should be available, functional, and reusable, such that other researchers can benefit from the work, verify the claims of the paper, and then build upon the software to do more great work. These ideals are summarised by the FAIR principles of data, which can be applied to software: research software should be Findable, Accessible, Interoperable, and Reusable. In order to help promote [FAIR software](https://www.nature.com/articles/s41597-022-01710-x), Climate Informatics is embarking, for the first time, on an _Artefact Evaluation phase_ for full paper submissions, after those submissions have been accented for publication as the conference proceedings in [Environmental Data Science](https://www.cambridge.org/core/journals/environmental-data-science) following the traditional peer-review process. Artefact Evaluation provides an opportunity to embed the values of reproducibility into the publication process in a lightweight opt-in fashion, thus encouraging authors to make software available and the results of the paper reproducible.

Reviewer: Include the suggestion of the previous work in 2023 Reproducibility Challenge.

```suggestion
Climate Informatics, like many other communities and fields, has software at its heart. Underlying most publications is a novel piece of software playing some critical role, e.g., embodying a model, processing or analysing data, or producing a visualisation. In order for such software artefacts to have the most impact, they should be available, functional, and reusable, such that other researchers can benefit from the work, verify the claims of the paper, and then build upon the software to do more great work. These ideals are summarised by the FAIR principles of data, which can be applied to software: research software should be Findable, Accessible, Interoperable, and Reusable. In order to help promote [FAIR software](https://www.nature.com/articles/s41597-022-01710-x) and following the positive outcomes of the [2023 Reproducibility Challenge](https://doi.org/10.5281/zenodo.1037613), Climate Informatics is embarking, for the first time, on an _Artefact Evaluation phase_ for full paper submissions, after those submissions have been accented for publication as the conference proceedings in [Environmental Data Science](https://www.cambridge.org/core/journals/environmental-data-science) following the traditional peer-review process. Artefact Evaluation provides an opportunity to embed the values of reproducibility into the publication process in a lightweight opt-in fashion, thus encouraging authors to make software available and the results of the paper reproducible.
```

# Guidelines for Reviewers

Reviewers will be recruited to the AEC from the [Climate Informatics community](http://www.climateinformatics.org) (join [here](https://groups.google.com/g/climate-informatics-news)) and [Turing Environment and Sustainability Grand Challenge community](https://cassgvp.kumu.io/alan-turing-institute-environment-and-sustainability) (join [here](https://forms.office.com/pages/responsepage.aspx?id=p_SVQ1XklU-Knx-672OE-ZmEJNLHTHVFkqQ97AaCfn9UMTZKT1IwTVhJRE82UjUzMVE2MThSOU5RMC4u)). Please join either (or both!) communities to receive updates on reviewer training opportunities.

## Requirements for reviewers
We require a minimum level of practical experience in climate data science to participate as a reviewer. This ensures that the review process can focus on technical detail rather than computational literacy. Reviewers will be required to evidence their experience through their own software publications. Full reviewer requirements will be published ==XXX==

## Expected workload

> - Each reviewer will be assigned 2 submissions

Reviewer: ```suggestion
- Each reviewer will be assigned 2 submissions.
```

> - 2-hour on-boarding session to get everyone up to speed

Reviewer: ```suggestion
- 2-hour on-boarding session to get everyone up to speed.
```

> - Each review can be expected to take ==XXX== hours to complete, including direct communications with the submitting author via our review portal

Reviewer: ```suggestion
- Each review can be expected to take ==XXX== hours to complete, including direct communications with the submitting author via our review portal.
```

> - Final reviews should be submitted by ==XXX==

Reviewer: ```suggestion
- Final reviews should be submitted by ==XXX==.
```

##  Evaluation process

Reviewers will asses the artefacts against a checklists provided for each ["badge" level to be awarded](https://github.com/alan-turing-institute/climate-informatics-2024-ae/blob/main/badges.md).

For each point, reviewers will be required to briefly note the evidence for this point, anything they have done to validate that point (e.g., what did you need to reproduce a claim), as well as the outcome (negative, neutral, or positive) and a brief reason for their judgment.

## Feedback to the authors

This process has been designed to be collaborative and iterative. During the first 3 weeks, reviewers will be able to feed-back to authors.

> Authors will be encouraged to push improvements to the artefact whenever possible. For example, if essential documentation is missing, then rather than submitting a negative evaluation or providing only the reviewers with the required information, the authors should provide that information via a versioned updated to the artefact.

Reviewer: ```suggestion
Authors will be encouraged to push improvements to the artefact whenever possible. For example, if essential documentation is missing, then rather than submitting a negative evaluation or providing only the reviewers with the required information, the authors should provide that information via a versioned/updated artefact.
```

# Author submission guidelines

## What authors should provide in their artefact submission

> Artifact submissions require three parts, as described below. ==Authors will be invited to attend reviewer training, to receive full guidance on the expectations and processes for submission described below.==

Reviewer: ```suggestion
Artefact submissions require three parts, as described below. ==Authors will be invited to attend reviewer training, to receive full guidance on the expectations and processes for submission described below.==
```

1. An overview of the artifact.

> 2. A non-institutional URL pointing to either: a single file containing the artifact (recommended), or the address of a public source control repository. The URL must be non-institutional to protect the anonymity of reviewers. Acceptable URLs can be obtained from Google Drive, Dropbox, Gitlab, Zenodo, and other providers. URLs may be generated using reviewer-only links. ==Do we prefer a versioned URL (e.g. DOI) here, or is that only expected if "accessible"==

Reviewer: ```suggestion
2. A non-institutional URL pointing to either: a single file containing the artefact (recommended), or the address of a public source control repository. The URL must be non-institutional to protect the anonymity of reviewers. Acceptable URLs can be obtained from Google Drive, Dropbox, Gitlab, Zenodo, and other providers. URLs may be generated using reviewer-only links. ==Do we prefer a versioned URL (e.g. DOI) here, or is that only expected if "accessible"==
```

Reviewer: I suggest we avoid recommending Google Drive and/or Dropbox. Instead, we should promote accepted version control platforms for software and data.

> 3. A hash certifying the version of the artifact at submission time: either an md5 hash of the single file (use the md5 or md5sum command-line tool to generate the hash), or the full commit hash for the repository (e.g., from git reflog --no-abbrev).

Reviewer: ```suggestion
3. A hash certifying the version of the artefact at submission time: either an md5 hash of the single file (use the md5 or md5sum command-line tool to generate the hash), or the full commit hash for the repository (e.g., from git reflog --no-abbrev).
```

> Artifacts do not need to be anonymous as reviewers will be aware of author identities.

Reviewer: ```suggestion
Artefacts do not need to be anonymous as reviewers will be aware of author identities.
```

>     - For unsupported claims, explain why they are omitted. For example, this can be reasonable if it requires very long runs on clusters, or involves data sets that are proprietary. In this case, the authors should provide some mechanism by which claims can be tested by some other mechanism to establish the veracity of the approach, e.g., a smaller run or a different sample data set, with expected outputs.

Reviewer: ```suggestion
    - For unsupported claims, explain why they are omitted. For example, this can be reasonable if it requires very long runs on clusters, or involves data sets that are proprietary. In this case, the authors should provide some mechanism by which claims can be tested to establish the veracity of the approach, e.g., a smaller run or a different sample data set, with expected outputs.
```

> A good way to package artifacts is as a virtual machine (VM). VMs give an easily reproducible environment that is somewhat resistant to bit rot. They also give reviewers confidence that errors or other problems cannot cause harm to their machines. Source code artifacts should use Docker or another build tool to manage all compilation and dependencies. This improves the odds that the reviewers will be able to quickly and painlessly install the artifact — without getting lost in environment issues ([e.g. what Python do I need?!](https://xkcd.com/1987/)).

Reviewer: ```suggestion
A good way to package artefacts is as a virtual machine (VM). VMs give an easily reproducible environment that is somewhat resistant to bit rot. They also give reviewers confidence that errors or other problems cannot cause harm to their machines. Source code artefacts should use Docker or another build tool to manage all compilation and dependencies. This improves the odds that the reviewers will be able to quickly and painlessly install the artifact — without getting lost in environment issues ([e.g. what Python do I need?!](https://xkcd.com/1987/)).
```

# Support

> There will be a 1-hour informal session for authors ==and reviewers== who are considering submitting artefacts to give an overview of the approach. This will happen online after CI 2024.

Reviewer: ```suggestion
There will be a 1-hour informal session for authors ==and reviewers== who are considering submitting artefacts to give an overview of the approach. This will happen online after CI2024 and informed via the dedicated [webpage for AE](https://alan-turing-institute.github.io/climate-informatics-2024/artefact-evaluation/) in the conference website.
```
